src/base/search/searchpluginmanager.cpp

```diff
@@ -27,73 +27,76 @@
-    PL_LEECHS,
-    PL_ENGINE_URL,
-    PL_DESC_LINK,
-    NB_PLUGIN_COLUMNS
-};
+#include "base/preferences.h"
+#include "base/profile.h"
+#include "base/utils/fs.h"
+#include "base/utils/misc.h"
+#include "searchdownloadhandler.h"
+#include "searchhandler.h"
 
-static inline void removePythonScriptIfExists(const QString &scriptPath)
+namespace
 {
-    Utils::Fs::forceRemove(scriptPath);
-    Utils::Fs::forceRemove(scriptPath + "c");
+    inline void removePythonScriptIfExists(const QString &scriptPath)
+    {
+        Utils::Fs::forceRemove(scriptPath);
+        Utils::Fs::forceRemove(scriptPath + "c");
+    }
 }
 
-const QHash<QString, QString> SearchEngine::m_categoryNames = SearchEngine::initializeCategoryNames();
+QPointer<SearchPluginManager> SearchPluginManager::m_instance = nullptr;
+
+const QHash<QString, QString> SearchPluginManager::m_categoryNames {
+    {"all", QT_TRANSLATE_NOOP("SearchEngine", "All categories")},
+    {"movies", QT_TRANSLATE_NOOP("SearchEngine", "Movies")},
+    {"tv", QT_TRANSLATE_NOOP("SearchEngine", "TV shows")},
+    {"music", QT_TRANSLATE_NOOP("SearchEngine", "Music")},
+    {"games", QT_TRANSLATE_NOOP("SearchEngine", "Games")},
+    {"anime", QT_TRANSLATE_NOOP("SearchEngine", "Anime")},
+    {"software", QT_TRANSLATE_NOOP("SearchEngine", "Software")},
+    {"pictures", QT_TRANSLATE_NOOP("SearchEngine", "Pictures")},
+    {"books", QT_TRANSLATE_NOOP("SearchEngine", "Books")}
+};
 
-SearchEngine::SearchEngine()
+SearchPluginManager::SearchPluginManager()
     : m_updateUrl(QString("http://searchplugins.qbittorrent.org/%1/engines/").arg(Utils::Misc::pythonVersion() >= 3 ? "nova3" : "nova"))
-    , m_searchStopped(false)
 {
-    updateNova();
-
-    m_searchProcess = new QProcess(this);
-    m_searchProcess->setEnvironment(QProcess::systemEnvironment());
-    connect(m_searchProcess, &QProcess::started, this, &SearchEngine::searchStarted);
-    connect(m_searchProcess, &QProcess::readyReadStandardOutput, this, &SearchEngine::readSearchOutput);
-    connect(m_searchProcess, static_cast<void (QProcess::*)(int)>(&QProcess::finished), this, &SearchEngine::processFinished);
-
-    m_searchTimeout = new QTimer(this);
-    m_searchTimeout->setSingleShot(true);
-    connect(m_searchTimeout, &QTimer::timeout, this, &SearchEngine::onTimeout);
+    Q_ASSERT(!m_instance); // only one instance is allowed
+    m_instance = this;
 
+    updateNova();
     update();
 }
 
-SearchEngine::~SearchEngine()
+SearchPluginManager::~SearchPluginManager()
 {
     qDeleteAll(m_plugins);
-    cancelSearch();
 }
 
-QStringList SearchEngine::allPlugins() const
+SearchPluginManager *SearchPluginManager::instance()
+{
+    return m_instance;
+}
+
+QStringList SearchPluginManager::allPlugins() const
 {
     return m_plugins.keys();
 }
 
-QStringList SearchEngine::enabledPlugins() const
+QStringList SearchPluginManager::enabledPlugins() const
 {
     QStringList plugins;
     foreach (const PluginInfo *plugin, m_plugins.values()) {
```

Comment:
You can remove the curly braces from this `foreach`.

Reply:
@Chocobo1 this

Reply:
The code in curly braces have multiple lines, I would prefer keeping the braces than remove it.

Reply:
Don't forget about the other comment :)